==============================
doctest for 7-base_geometry.py
==============================

>>> BaseGeometry = __import__('7-base_geometry').BaseGeometry
>>> bg = BaseGeometry()

TEST WITH SUCCESS
===============

testing main test
::
>>> print(dir(bg))
['__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', 'area', 'integer_validator']

testing from main 6
::
>>> try:
...     print(bg.area())
... except Exception as e:
...     print(e)
area() is not implemented

testing from main 7
::
>>> try:
...     bg.integer_validator("name", "John")
... except Exception as e:
...     print(e)
name must be an integer

testing with 0
::
>>> try:
...     bg.integer_validator("age", 0)
... except Exception as e:
...     print(e)
age must be greater than 0

testing with a negativ number
::
>>> try:
...     bg.integer_validator("distance", -4)
... except Exception as e:
...     print(e)
distance must be greater than 0

testing validator with a float
::
>>> try:
...     bg.integer_validator("baptiste", 3.14)
... except Exception as e:
...     print(e)
baptiste must be an integer

testing validator with a dictionnary
::
>>> try:
...     bg.integer_validator("baptiste", {cestmoi, cestlui})
... except Exception as e:
...     print(e)
name 'cestmoi' is not defined

testing validator with a tulpe
::
>>> try:
...     bg.integer_validator("baptiste", (0, 1, 2))
... except Exception as e:
...     print(e)
baptiste must be an integer

testing validator with a none as name and value
::
>>> try:
...     bg.integer_validator(None, None)
... except Exception as e:
...     print(e)
None must be an integer

testing validator with 3 arguments
::
>>> try:
...     bg.integer_validator("baptiste", "encore", "un", "baptiste")
... except Exception as e:
...     print(e)
BaseGeometry.integer_validator() takes 3 positional arguments but 5 were given

testing validator with no arguments
::
>>> try:
...     bg.integer_validator()
... except Exception as e:
...     print(e)
BaseGeometry.integer_validator() missing 2 required positional arguments: 'name' and 'value'

testing with a succed
::
>>> try:
...     bg.integer_validator("baptiste", 32)
... except Exception as e:
...     print(e)